SofascoreAcademy/SofascoreAcademy/LeagueView.swift

```diff
@@ -0,0 +1,88 @@
+//
+//  LeagueView.swift
+//  SofascoreAcademy
+//
+//  Created by Akademija on 12.03.2024..
+//
+
+import Foundation
+import UIKit
+import SofaAcademic
+
+class LeagueView: BaseView {
+    
+    private let stackView = UIStackView()
+    private let leagueInfoView = LeagueInfoView()
+    private var matchViews: [MatchView] = []
+    
+    func updateScore(matchId: Int, score: Int, side: teamSide) {
+        let matchToChange = matchViews.first(where: {
+            $0.matchId == matchId
+        })
+        matchToChange?.updateScore(score: score, side: side)
+    }
+    
+    func updateMatchStatus(matchId: Int, status: matchStatus) {
+        let matchToChange = matchViews.first(where: {
+            $0.matchId == matchId
+        })
+        matchToChange?.updateMatchStatus(status: status)
+    }
+    
+    func updateMatchTime(matchId: Int, time: Int) {
+        let matchToChange = matchViews.first(where: {
+            $0.matchId == matchId
+        })
+        matchToChange?.updateMatchTime(time: time)
+    }
+    
+    func update(data: [matchData]){
+        leagueInfoView.update(countryName: "Spain", leagueName: "LaLiga", leagueLogo: "leagueLogo")
+        for matchData in data {
+            let matchView = MatchView()
+            matchView.update(data: matchData)
+            matchViews.append(matchView)
+        }
+        
+        addViews()
+        styleViews()
+        setupConstraints()
+    }
+    
+    override init() {
+        super.init()
+    }
+    
+    override func addViews() {
+        addSubview(stackView)
+        
+        for matchView in matchViews {
+            stackView.addArrangedSubview(matchView)
+        }
+    
+        addSubview(leagueInfoView)
+    }
+    
+    override func styleViews() {
+        stackView.axis = .vertical
+    }
+
+    override func setupConstraints() {
+        leagueInfoView.snp.makeConstraints() {
+            $0.leading.trailing.equalToSuperview()
+            $0.top.bottom.equalToSuperview()
+        }
+        
+        stackView.snp.makeConstraints {
+            $0.top.equalTo(leagueInfoView.snp.bottom)
+            $0.leading.trailing.equalToSuperview()
+        }
+        
+        for (_, matchView) in matchViews.enumerated() {                    
+            matchView.snp.makeConstraints {
+                $0.height.equalTo(56)
```

Comment:
Bilo bi dobro da se sve komponente same izracunaju. Tocnije ovaj consraint mozes prebaciti unutar samog matchview. Kad programiramo UI, razlazemo stvari na smislene male komponente koje su reusabilne. Dakle, zelimo ih koristiti po potrebi na vise mjesta, a ne da uvijek sve pismo ispocetka. I sad na ovaj nacin gdje god cemo koristiti matchview trebat cemo mu reci height i dizajner promjeni height i ti sad moras trazit po cijelom appu gdje se to koristi, puno lakse je promjeniti height na samo jednom mjestu.